### PR TITLE
Allow admins to fulfill and ship orders

### PR DESCRIPTION
**Pull request by johnktravers**
Hey! The main functionality that I added was allowing admins to access merchant order pages and fulfill and ship orders. 

I also added the paths and views for admins to access the user orders index and user orders show pages. Let me know what you think!

### CODE REVIEW

File: app/controllers/admin/merchant_orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    item.reduce_inventory(item_order.quantity)
+    item.save
+
+    order = item_order.order
+    if order.item_orders.all? { |item_ord| item_ord.fulfilled? }
+      order.update(status: 0)
+    end
+
+    flash[:success] = ["You have successfully fulfilled #{item.name} for Order ##{params[:order_id]}"]
+    redirect_to "/admin/merchants/#{params[:merchant_id]}/orders/#{params[:order_id]}"
```
Comment (zacisaacson) [inline app/controllers/admin/merchant_orders_controller.rb:27]: Looks great!

File: app/controllers/merchant/orders_controller.rb [hunk 2]
```diff
@@ -17,6 +17,11 @@ def update
     item.reduce_inventory(item_order.quantity)
     item.save
 
+    order = item_order.order
+    if order.item_orders.all? { |item_ord| item_ord.fulfilled? }
+      order.update(status: 0)
```
Comment (zacisaacson) [inline app/controllers/merchant/orders_controller.rb:22]: I like this all? enumerable!

File: app/views/admin/merchant_orders/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<%= render partial: 'partials/merchant_order_show', locals:
```
Comment (zacisaacson) [inline app/views/admin/merchant_orders/show.html.erb:1]: Partials, yes!

File: app/views/admin/user_orders/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<%= render partial: 'partials/user_orders_index', locals:
```
Comment (zacisaacson) [inline app/views/admin/user_orders/index.html.erb:1]: More partials, yes!

File: app/controllers/admin/merchant_orders_controller.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+class Admin::MerchantOrdersController < Admin::BaseController
+  def show
+    merchants = Merchant.where(id: params[:merchant_id])
+    orders = Order.where(id: params[:order_id])
+    if merchants.empty? || orders.empty?
+      render_404
```
Comment (grwthomps) [inline app/controllers/admin/merchant_orders_controller.rb:6]: I like this 404 helper method, really nice

File: app/controllers/admin/merchant_orders_controller.rb (hunk 1 above)
Comment (grwthomps) [reply app/controllers/admin/merchant_orders_controller.rb:27]: This update method is very clean

File: app/views/partials/_merchant_order_show.html.erb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                <%= button_to 'Fulfill Item', fulfill_item_path + "#{item_order.id}", method: :patch, class: 'btn btn-outline-success' %>
+              <% else %>
+                <p>Cannot Fulfill <br>(Not Enough Inventory)</p>
+              <% end %>
+            <% elsif item_order.fulfilled? %>
+              <p class="active">Fulfilled</p>
+            <% elsif item_order.cancelled? %>
+              <p class="inactive">Cancelled</p>
+            <% end %>
+          </td>
```
Comment (zacisaacson) [inline app/views/partials/_merchant_order_show.html.erb:49]: Looks awesome, great readability!

**Review by zacisaacson — APPROVED** [review]

File: db/seeds.rb [hunk 7]
```diff
@@ -166,25 +166,30 @@
...
-order_5.item_orders.create(item_id: lily.id, quantity: 3, price: lily.price)
-order_5.item_orders.create(item_id: dahlia.id, quantity: 4, price: dahlia.price)
+order_5 = lynda.orders.create(status: 0)
+order_5.item_orders.create(item_id: lily.id, quantity: 3, price: lily.price, status: 1)
+order_5.item_orders.create(item_id: dahlia.id, quantity: 4, price: dahlia.price, status: 1)
+
+order_6 = cynthia.orders.create
+order_6.item_orders.create(item_id: lily.id, quantity: 2, price: maple.price)
+order_6.item_orders.create(item_id: tulip.id, quantity: 3, price: tulip.price)
+order_6.item_orders.create(item_id: rose.id, quantity: 10, price: tulip.price)
```
Comment (grwthomps) [inline db/seeds.rb:192]: Thanks for the extra seeds!

**Review by grwthomps — APPROVED** [review]

**Pull request merged**